### PR TITLE
Extend Framework

### PR DESCRIPTION
**Pull request by Koenari**
I would appreciate someone really looking this over (especially for typos and accidental api breakage)

Adds following classes
- ThreadManager
- Cursor
- GameWindow *
- File-Manager/-Thread-AccessPath

\* Acknowledgements:
- GameWindow is a completely assumed name and most likely is a much more low-level class

### CODE REVIEW

File: FFXIVClientStructs/FFXIV/Client/System/File/FileAccessPath.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    [FieldOffset(0x208)] public char* LongStringPtr;
+
+    public string AsString {
+        get {
+            if (LongStringPtr == null)
+                fixed (char* p = Buffer)
+                    return new string(p);
+            else return new string(LongStringPtr);
+        }
+    }
```
Comment (Haselnussbomber) [inline FFXIVClientStructs/FFXIV/Client/System/File/FileAccessPath.cs:15]: I'd suggest overriding C#'s `ToString()` instead.
```suggestion
    public override string ToString() {
        if (LongStringPtr == null)
            fixed (char* p = Buffer)
                return new string(p);
        else
            return new string(LongStringPtr);
    }
```

**Review by Haselnussbomber — COMMENTED** [review]
Couple suggestions from me. 🙂

File: ida/data.yml [hunk 2]
```diff
@@ -2233,6 +2261,17 @@ classes:
...
+    vtbls:
+      - ea: 0x1419B0378
+    vfuns:
+      0: dtor
+      4: SetCursorTyp
+      5: SetIsCursorVisible
+      8: SetUSeSoftwareCursor
+      9: GetUSeSoftwareCursor
+      14: IsCursorVisible
+      21: GetIsMouseNotCaptured
```
[the remark this answers is not in the document]
Comment (Haselnussbomber) [reply ida/data.yml:2274]: Edited this. Autopilot brain renamed them to Enable/DisableSoftwareCursor, but they aren't doing that, sorry. 🤦‍♂️
Fixed the typo though. 🙂

File: FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs [hunk 3]
```diff
@@ -39,7 +51,9 @@ public unsafe partial struct Framework {
     [FieldOffset(0x17C4)] public float FrameRate;
     [FieldOffset(0x17D0)] public bool WindowInactive;
 
-    [FieldOffset(0x220C)] private fixed char userPath[260]; // WideChar Array
+    [FieldOffset(0x19EC)] private fixed char gamePath[260];
+    [FieldOffset(0x1DFC)] private fixed char sqPackPath[260];
+    [FieldOffset(0x220C)] private fixed char savePath[260];
```
[the remark this answers is not in the document]
Comment (Haselnussbomber) [reply FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs:56]: Oh, it seems like these are UTF16 and the FixedStringGenerator only supports UTF8, so this won't work.

File: ida/data.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Koenari) [reply ida/data.yml]: Yeah those two are jsut Get/Set for the field

File: FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs [hunk 4]
```diff
@@ -65,11 +81,26 @@ public unsafe partial struct Framework {
     [MemberFunction("E8 ?? ?? ?? ?? 89 47 2C")]
     public static partial long GetServerTime();
 
-    public string UserPath {
-        get {
-            fixed (char* p = userPath) {
-                return new string(p);
-            }
+    [Obsolete("Use SavePath")]public string UserPath => SavePath;
```
Comment (Caraxi) [inline FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs:84]: Why rename it? The launch parameter it is set by is called UserPath

Comment (Koenari) [reply FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs]: I did miss that and opted for a more descriptive name. Is reverted now as you are correct and we should adhere to SE's naming

File: FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs [hunk 5]
```diff
@@ -50,6 +64,8 @@ public unsafe partial struct Framework {
 
     [FieldOffset(0x2BF0)] public GameVersion GameVersion;
 
+    [FieldOffset(0x35C0)] public void* SteamApiLibrary;
+
```
Comment (Haselnussbomber) [inline FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs:68]: `SteamApiLibrary` can be omitted, PR #748 is going to add this. 🙂
```suggestion
```

Comment (Koenari) [reply FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs:68]: Not hte same field, this is the DLL 0x35B8 is the wrapper

Comment (Haselnussbomber) [reply FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs:68]: I see, there is a mistake in the other PR. Sorry.

**Pull request merged**